### DISCUSSION TITLE
[Bug]: After loadFromJSON getActiveObject() return undefined

### DISCUSSION THREAD

**Discussion by Saqib92** [fabricjs/fabric.js#8646] [Q&A] [answered]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

5.2.1

### In What environments are you experiencing the problem?

Firefox, Chrome, Safari, Opera, Microsoft Edge

### Node Version (if applicable)

None

### Link To Reproduction

https://stackblitz.com/edit/angular-ivy-svswdu?file=src%2Fapp%2Fapp.component.html,src%2Fapp%2Fapp.component.ts

### Steps To Reproduce

Save to JSON:
```
let myJson = JSON.stringify(this.canvas.toJSON([
  'id',
  'cornerColor',
  'cornerStyle',
  'borderColor',
  'hasBorders',
  'cornerSize',
  'centeredScaling',
  'transparentCorners',
  'hasControls',
  'erasable',
  '_controlsVisibility'
]));
```

Load from JSON:

```
let j = JSON.parse(myJson);
this.canvas.loadFromJSON(j, () => {
  this.canvas.renderAll();
});
```

## GO TO ANOTHER ROUTE 
this looks like issue.

Loading json after coming from other page:

```
let canvasObjects = this.canvas.getActiveObject();
console.log(canvasObjects);  // undefined
```

### Expected Behavior

Should give active object after loading from JSON. 

### Actual Behavior

if i save json and go to other page and come back and load json. `getActiveObject()` Not returning active object after loading from JSON.

**Comment by ShaMan123** [comment, marked as answer]
This is expected behavior.
You can use `setActiveObject`, no object is active after loading from json

**Reply by Saqib92** [reply]
Will try this, but i am clicking on object before calling `getActiveObject()` it should make object active on clicking.

**Reply by Saqib92** [reply]
There is something wrong. 
If i go to pageA and Comes back. it working, 
If i go to PageB and comes back . Its Not working. 

```
let j = JSON.parse(myJson);
this.canvas.loadFromJSON(j, () => {
  this.canvas.renderAll();
});
const co = this.canvas.getObjects();
co.forEach((obj: any) => {
  this.canvas.setActiveObject(obj); // activating all objects. 
});
```

**Reply by ShaMan123** [reply]
are you aware that loading is async?
